backend/ipp.c

```diff
@@ -3367,9 +3368,24 @@ run_as_user(char       *argv[],		/* I - Command-line arguments */
   xpc_dictionary_set_fd(request, "stderr", 2);
   xpc_dictionary_set_fd(request, "side-channel", CUPS_SC_FD);
 
-  response = xpc_connection_send_message_with_reply_sync(conn, request);
+  sem      = dispatch_semaphore_create(0);
+  response = NULL;
 
+  xpc_connection_send_message_with_reply(conn, request,
+                                         dispatch_get_global_queue(0,0),
+					 ^(xpc_object_t reply)
+					 {
+					   /* Save the response and wake up */
+					   if (xpc_get_type(reply)
+					           == XPC_TYPE_DICTIONARY)
+					     response = xpc_retain(reply);
+
+					   dispatch_semaphore_signal(sem);
+					 });
+
+  dispatch_semaphore_wait(sem, DISPATCH_TIME_FOREVER);
   xpc_release(request);
+  dispatch_release(sem);
```

Comment:
xpc_connection_send_message_with_reply_sync is supported on the same exact Os versions as xpc_connection_send_message_with_reply. Apple recommends the former, hence why I switched to it.